modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinSpringServerCodegen.java

```diff
@@ -0,0 +1,442 @@
+                "Useful to help facilitate the generation gap pattern", serviceInterface);
+        addSwitch(SERVICE_IMPLEMENTATION, "generate stub service implementations that extends service " +
+                "interfaces. If this is set to true service interfaces will also be generated", serviceImplementation);
+        addSwitch(USE_BEANVALIDATION, "Use BeanValidation API annotations to validate data types", useBeanValidation);
+
+        supportedLibraries.put(SPRING_BOOT, "Spring-boot Server application.");
+        setLibrary(SPRING_BOOT);
+
+        CliOption cliOpt = new CliOption(CodegenConstants.LIBRARY, "library template (sub-template) to use");
+        cliOpt.setDefault(SPRING_BOOT);
+        cliOpt.setEnum(supportedLibraries);
+        cliOptions.add(cliOpt);
+    }
+
+    public String getResourceFolder() {
+        return this.resourceFolder;
+    }
+
+    public void setResourceFolder(String resourceFolder) {
+        this.resourceFolder = resourceFolder;
+    }
+
+    public String getBasePackage() {
+        return this.basePackage;
+    }
+
+    public void setBasePackage(String basePackage) {
+        this.basePackage = basePackage;
+    }
+
+    public String getServerPort() {
+        return this.serverPort;
+    }
+
+    public void setServerPort(String serverPort) {
+        this.serverPort = serverPort;
+    }
+
+    public boolean getExceptionHandler() {
+        return this.exceptionHandler;
+    }
+
+    public void setExceptionHandler(boolean exceptionHandler) {
+        this.exceptionHandler = exceptionHandler;
+    }
+
+    public boolean getGradleBuildFile() {
+        return this.gradleBuildFile;
+    }
+
+    public void setGradleBuildFile(boolean gradleBuildFile) {
+        this.gradleBuildFile = gradleBuildFile;
+    }
+
+    public boolean getSwaggerAnnotations() {
+        return this.swaggerAnnotations;
+    }
+
+    public void setSwaggerAnnotations(boolean swaggerAnnotations) {
+        this.swaggerAnnotations = swaggerAnnotations;
+    }
+
+    public boolean getServiceInterface() {
+        return this.serviceInterface;
+    }
+
+    public void setServiceInterface(boolean serviceInterface) {
+        this.serviceInterface = serviceInterface;
+    }
+
+    public boolean getServiceImplementation() {
+        return this.serviceImplementation;
+    }
+
+    public void setServiceImplementation(boolean serviceImplementation) {
+        this.serviceImplementation = serviceImplementation;
+    }
+
+    public boolean getUseBeanValidation() {
+        return this.useBeanValidation;
+    }
+
+    @Override
+    public void setUseBeanValidation(boolean useBeanValidation) {
+        this.useBeanValidation = useBeanValidation;
+    }
+
+    @Override
+    public CodegenType getTag() {
+        return CodegenType.SERVER;
+    }
+
+    @Override
+    public String getName() {
+        return "kotlin-spring";
```

Comment:
If it's desired overall to have this as a library under `kotlin-server`, id be happy to make those changes. However, I'd probably continue to extend `AbstractKotlinCodegen`, rather than extending `KotlinServerCodegen` (if possible).